core/templates/services/playthrough-issues-backend-api.service.spec.ts

```diff
@@ -50,81 +45,66 @@ describe('PlaythroughIssuesBackendApiService', function() {
     is_valid: true
   }];
 
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector, $q) {
-    PlaythroughIssuesBackendApiService =
-      $injector.get('PlaythroughIssuesBackendApiService');
-    $httpBackend = $injector.get('$httpBackend');
-    PlaythroughIssueObjectFactory = $injector.get(
-      'PlaythroughIssueObjectFactory');
-    PlaythroughObjectFactory = $injector.get('PlaythroughObjectFactory');
-
-    CsrfService = $injector.get('CsrfTokenService');
-
-    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
-      var deferred = $q.defer();
-      deferred.resolve('sample-csrf-token');
-      return deferred.promise;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
     });
-  }));
+    httpTestingController = TestBed.get(HttpTestingController);
+    playthroughIssuesBackendApiService = TestBed.get(
+      PlaythroughIssuesBackendApiService);
+    playthroughIssueObjectFactory = TestBed.get(PlaythroughIssueObjectFactory);
+  });
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
-  describe('.fetch', function() {
-    it('returns the issues data provided by the backend', function() {
-      var successHandler = jasmine.createSpy('success');
-      var failureHandler = jasmine.createSpy('failure');
-      $httpBackend.expectGET(
-        '/issuesdatahandler/7?exp_version=1'
-      ).respond(backendIssues);
+  describe('.fetch', () => {
+    it('returns the issues data provided by the backend', fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failureHandler = jasmine.createSpy('failure');
 
-      PlaythroughIssuesBackendApiService.fetchIssues('7', 1).then(
+      playthroughIssuesBackendApiService.fetchIssues('7', 1).then(
         successHandler, failureHandler);
-      $httpBackend.flush();
 
-      expect(successHandler).toHaveBeenCalledWith(
-        backendIssues.map(PlaythroughIssueObjectFactory.createFromBackendDict));
+      let req = httpTestingController.expectOne(
+        '/issuesdatahandler/7?exp_version=1');
+      expect(req.request.method).toEqual('GET');
+      req.flush(backendIssues);
+
+      flushMicrotasks();
+      expect(successHandler).toHaveBeenCalled();
```

Comment:
Do not replace `toHaveBeenCalledWith` with `toHaveBeenCalled`.

Reply:
But the example given over here at https://github.com/oppia/oppia/wiki/Angular-Migration is doing the same in this file as well topic-viewer-backend-api.service.spec.ts.

If I don't have to replace it then the parameters also should be present inside toHaveBeenCalledWith?

Reply:
Yes, the parameters should also be present. Please revert back this to `toHaveBeenCalledWith`.